pipelines/batch/src/main/java/com/google/fhir/analytics/FhirSearchUtil.java

```diff
@@ -317,6 +332,88 @@ Bundle searchByPatientAndLastDate(
     return query.execute();
   }
 
+  /**
+   * Fetch the details of bulk export job for the given bulkExportStatusUrl
+   *
+   * @param bulkExportStatusUrl the url of the bulk export job
+   * @return BulkExportHttpResponse - the status and details of the bulk export job
+   * @throws IOException
+   */
+  public BulkExportHttpResponse fetchBulkExportHttpResponse(String bulkExportStatusUrl)
+      throws IOException {
+    IHttpResponse httpResponse = fetchUtil.fetchResponseForUrl(bulkExportStatusUrl);
+    BulkExportHttpResponse bulkExportHttpResponse = new BulkExportHttpResponse();
+    bulkExportHttpResponse.setHttpStatus(httpResponse.getStatus());
+    if (httpResponse.getHeaders("expires") != null
+        && !httpResponse.getHeaders("expires").isEmpty()) {
+      String expiresString = httpResponse.getHeaders("expires").get(0);
+      Date expires = new Date(expiresString);
+      bulkExportHttpResponse.setExpires(expires);
+    }
+    if (!CollectionUtils.isEmpty(httpResponse.getHeaders("retry-after"))) {
+      String retryHeaderString = httpResponse.getHeaders("retry-after").get(0);
+      bulkExportHttpResponse.setRetryAfter(Integer.valueOf(retryHeaderString));
+    }
+    if (!CollectionUtils.isEmpty(httpResponse.getHeaders("x-progress"))) {
+      String xProgress = httpResponse.getHeaders("x-progress").get(0);
+      bulkExportHttpResponse.setXProgress(xProgress);
+    }
+
+    String body = null;
+    try (Reader reader = httpResponse.createReader()) {
+      body = IOUtils.toString(reader);
+    }
+
+    if (!Strings.isNullOrEmpty(body)) {
+      ObjectMapper objectMapper = new ObjectMapper();
+      BulkExportResponse bulkExportResponse =
+          objectMapper.readValue(body, BulkExportResponse.class);
+      bulkExportHttpResponse.setBulkExportResponse(bulkExportResponse);
+    }
+    return bulkExportHttpResponse;
+  }
+
+  /**
+   * Start the bulk export for the given resourceTypes with the fhirVersionEnum
+   *
+   * @param resourceTypes the types which needs to be exported
+   * @param fhirVersionEnum the fhir version of the resources to be exported
+   * @return the absolute url via which the status and details of the job can be fetched
+   */
+  public String triggerBulkExportJob(List<String> resourceTypes, FhirVersionEnum fhirVersionEnum)
+      throws BulkExportException {
+    Map<String, List<String>> headers = new HashMap<>();
+    headers.put("accept", Arrays.asList("application/fhir+ndjson"));
```

Comment:
ditto for header constants; also when possible please use [Apache httpcore constants](https://javadoc.io/doc/org.apache.httpcomponents/httpcore/latest/constant-values.html) as we already depend on it.